src/components/Table/Table.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState } from "react"
+import { ColumnConfig, ColumnConfigs, WithID } from "./types"
+
+interface Props<Datum extends WithID> {
+  data: Datum[]
+  columnConfigs: ColumnConfigs<Datum>
+}
+
+export function Table<Datum extends WithID>(props: Props<Datum>) {
+  const { data: dataUnsorted, columnConfigs } = props
+  const [sortConfig, setSortConfig] = useState<ColumnConfig<Datum>>()
+  const [sortAsc, setSortAsc] = useState<boolean>(false)
+
+  const data = [...dataUnsorted].sort((a, b) => {
+    const valueA = sortConfig?.getValue?.(a)
+    const valueB = sortConfig?.getValue?.(b)
+    if (isSortable(valueA) && isSortable(valueB)) {
+      if (valueA > valueB) return sortAsc ? 1 : -1
+      if (valueA < valueB) return sortAsc ? -1 : 1
+    }
+    return 0
+  })
+
+  const handleSort = (columnConfig: ColumnConfig<Datum>) => () => {
+    if (columnConfig.sortable && columnConfig.getValue) {
+      if (columnConfig === sortConfig || !sortConfig) {
+        setSortAsc((value) => !value)
+      }
+      setSortConfig(columnConfig)
+    }
+  }
+
+  const headers = columnConfigs.map((columnConfig, i) => {
+    return (
+      <th
+        scope="col"
+        key={`th-${columnConfig.label}-${i}`}
+        onClick={handleSort(columnConfig)}
+      >
+        {columnConfig.label}
+        {columnConfig === sortConfig ? (sortAsc ? "↑" : "↓") : " "}
+      </th>
+    )
+  })
+
+  const rows = data.map((datum) => (
+    <tr key={`tr-${datum.id}`}>
+      {columnConfigs.map((columnConfig, i) => (
+        <td key={`tr-${datum.id}-td-${i}`}>{columnConfig.getValue?.(datum)}</td>
```

Comment:
It looks like there's only valuable output when `getValue` is defined, should it be required? What's the case for it being optional?

Reply:
In my initial thinking, I was trying to accomodate filler columns—but now that I've played with this a bit, I'm not seeing that as a likely scenario. Making these required will simplify the implementation and usage. Good catch! 👍